File: .codegen/templates/entity.ts
import {
  BaseEntity, 
  PrimaryGeneratedColumn,
  Column, 
  OneToOne,
  JoinColumn, 
  ManyToOne,
  ManyToMany,
  Entity
} from "typeorm"
import { OtherDomain } from "./OtherDomain"
import { EnumName } from "../enum/EnumName"

@Entity('domain_name')
export class DomainName extends BaseEntity {

  @PrimaryGeneratedColumn({name:"id_domain_name"})
  id: number

  @Column({name:"column_name"/* column_length *//* column_required *//* column_enum */})
  attributeName: string

  @OneToOne(() => TypeDomain)
  @JoinColumn()
  attributeName: OtherDomain

  @OneToMany(() => TypeDomain, (typeDomain) => typeDomain.domain)
  attributeName: OtherDomain[]

  @ManyToOne(() => TypeDomain, (typeDomain) => typeDomain.domains)
  attributeName: OtherDomain

  @ManyToMany(() => TypeDomain, typeDomain => typeDomain.domains)
  attributeName: OtherDomain[]

}

File: .codegen/templates/entity.java
package com.example.demo.domain.model;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Objects;
import java.util.ArrayList;
import java.util.List;
import java.time.*;
import com.example.demo.domain.enums.*;

@Entity
@Table(name = "DOMAIN_NAME")
public class DomainName implements Serializable {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column(name = "ATTRIBUTE")
	private String attribute;

	@Enumerated(EnumType.STRING)
	@Column(name = "ATTRIBUTE")
	private String attribute;

	@OneToOne
	private String attribute;

    @ManyToMany
	@JoinTable(name = "DOMAIN_OTHER",
			joinColumns = @JoinColumn(name = "OTHER_ID"),
			inverseJoinColumns = @JoinColumn(name = "DOMAIN_ID"))
    private List<TypeDomainName> attribute = new ArrayList<>();

    @ManyToMany(mappedBy = "MAPPED_BY")
    private List<TypeDomainName> attribute = new ArrayList<>();

    @ManyToOne()
    private TypeDomainName attribute;

    @OneToMany()
    private List<TypeDomainName> attribute = new ArrayList<>();

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getAttribute() {
		return attribute;
	}

	public void setAttribute(String attribute) {
		this.attribute = attribute;
	}

	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (o == null || getClass() != o.getClass()) return false;
		DomainName domainName = (DomainName) o;
		return id.equals(domainName.id);
	}

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public String toString() {
		return "DomainName{" +
			"id=" + id +
			'}';
	}

}
